dom_test.go

```diff
@@ -45,6 +45,25 @@ func TestInput(t *testing.T) {
 	}
 }
 
+func TestEmptyElement(t *testing.T) {
+	t.Parallel()
+
+	got :=
+		dom.Element(
+			"", dom.Attrs(),
+			dom.InnerText("Good "),
+			dom.B(dom.Attrs(), dom.InnerText("morning")),
+			dom.InnerText(", world!"),
+		)
+
+	want := template.HTML(`Good <b>morning</b>, world!`)
```

Comment:
So now `dom.Element("", dom.Attrs()` effectively lets us "join several dom.Node" and produce a html string

Alternatively (or additionally), we can add a new function `dom.Join` but that'll introduce new vocabulary?

Reply:
i feel `Join` is clearer than using `dom.Element` tho 🤔

Reply:
👍 `domutils.Join` can be the API to introduce but fixing `dom.Element` is still necessary for `domutils.Join` to be possible.

i.e. `domutils.Join(a, b)` produces a `dom.Node` and because of this fix, `domutils.Join(a, b).HTML()` would work as expected

Reply:
👍 `domutils.Join` can be the API to introduce but fixing `dom.Element` is still necessary for `domutils.Join` to be possible.

i.e. `domutils.Join(a, b)` produces a `dom.Node` and because of this fix, `domutils.Join(a, b).HTML()` would work as expected